### PR TITLE
Lambda reward Wrappers

### PR DESCRIPTION
**Pull request by gianlucadecola**
This PR adds jumpy reward wrappers for compatibility with Jax and NumPy environments.
Observation and action wrapper replacement are also ready but left out from this PR for easier review.

@pseudo-rnd-thoughts

### CODE REVIEW

File: gym/dev_wrappers/lambda_reward.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import jax.numpy as jnp
+import jumpy as jp
+import numpy as np
+
+import gym
+from gym.dev_wrappers import FuncArgType
+from gym.error import InvalidBound
+
+
+class lambda_reward_v0(gym.RewardWrapper):
```
Comment (RedTachyon) [inline gym/dev_wrappers/lambda_reward.py:13]: Please no snake_case class names. I'd even take LambdaReward_v0, or LambdaRewardV0 over this

Comment (gianlucadecola) [reply gym/dev_wrappers/lambda_reward.py:13]: Afaik it was a naming convention for the new dev_wrappers (also here: https://github.com/openai/gym/pull/2853). If we want to switch to standard camel case no problem for me

Comment (pseudo-rnd-thoughts) [reply gym/dev_wrappers/lambda_reward.py:13]: Sorry I forgot to mention that I think we are keeping the normal python style with the addition of `V0` at the end.
@RedTachyon What do you think is better:`LambdaReward_v0` or `LambdaRewardV0`

File: gym/wrappers/__init__.py [hunk 2]
```diff
@@ -1,5 +1,13 @@
 """Module of wrapper classes."""
 from gym import error
+
+try:
+    import jumpy
+except ImportError:
+    pass
+else:
+    from gym.dev_wrappers.lambda_reward import clip_rewards_v0, lambda_reward_v0
```
Comment (pseudo-rnd-thoughts) [inline gym/wrappers/__init__.py:9]: Could this be part of the import jumpy section?

File: gym/wrappers/__init__.py [hunk 3]
```diff
@@ -1,5 +1,13 @@
 """Module of wrapper classes."""
 from gym import error
+
+try:
+    import jumpy
+
+    from gym.dev_wrappers.lambda_reward import ClipRewardsV0, LambdaRewardV0
+except ImportError:
+    pass
```
Comment (pseudo-rnd-thoughts) [inline gym/wrappers/__init__.py:9]: What happens when this fails but the user tries to import the class? 
Is the error helpful?

Comment (gianlucadecola) [reply gym/wrappers/__init__.py:9]: Good catch, maybe raising a warning telling the user that all dev_wrappers will not work correctly since jumpy is not available for python3.6 would be better?

Comment (pseudo-rnd-thoughts) [reply gym/wrappers/__init__.py:9]: I think that python 3.6 will be removed for the new wrappers

Comment (gianlucadecola) [reply gym/wrappers/__init__.py:9]: If python3.6 supports removal is programmed anytime soon I can wait and then remove the whole try - except block from all the dev_wrappers PRs, what do you think? Removing the try-except block now will cause python3.6 pipelines to fail because of jumpy

**Comment by gianlucadecola** [comment]
Closed in favour of PR on Gymnasium
